### PR TITLE
Remove components if not extracted

### PR DESCRIPTION
**Pull request by akimakinai**
# Objective

- Fixes https://github.com/bevyengine/bevy/issues/15871
(Camera is done in #15946)

## Solution

- Do the same as #15904 for other extraction systems
- Added missing `SyncComponentPlugin` for DOF, TAA, and SSAO
(According to the [documentation](https://dev-docs.bevyengine.org/bevy/render/sync_component/struct.SyncComponentPlugin.html), this plugin "needs to be added for manual extraction implementations." We may need to check this is done.)

## Testing

Modified example locally to add toggles if not exist.
- [x] DOF - toggling DOF component and perspective in `depth_of_field` example
- [x] TAA - toggling `Camera.is_active` and TAA component
- [x] clusters - not entirely sure, toggling `Camera.is_active` in `many_lights` example (no crash/glitch even without this PR)
- [x] previous_view - toggling `Camera.is_active` in `skybox` (no crash/glitch even without this PR)
- [x] lights - toggling `Visibility` of `DirectionalLight` in `lighting` example
- [x] SSAO - toggling `Camera.is_active` and SSAO component in `ssao` example
- [x] default UI camera view - toggling `Camera.is_active` (nop without #15946 because UI defaults to some camera even if `DefaultCameraView` is not there)
- [x] volumetric fog - toggling existence of volumetric light. Looks like optimization, no change in behavior/visuals

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@rafalh are you able to review this PR for me in turn? :) This is closely related to #15946.

File: crates/bevy_pbr/src/render/light.rs [hunk 1]
```diff
@@ -379,6 +379,14 @@ pub fn extract_lights(
     ) in &directional_lights
     {
         if !view_visibility.get() {
+            commands
+                .get_entity(entity)
+                .expect("Light entity wasn't synced.")
+                .remove::<(
+                    ExtractedDirectionalLight,
+                    RenderCascadesVisibleEntities,
+                    LightViewEntities,
```
Comment (rafalh) [inline crates/bevy_pbr/src/render/light.rs:388]: `LightViewEntities` is added in an observer and not in this system, so it's not clear at the first glance why we remove it here. Maybe we should remove it in an observer too to be consistent? :thinking:

File: crates/bevy_core_pipeline/src/dof/mod.rs [hunk 2]
```diff
@@ -342,6 +346,7 @@ impl ViewNode for DepthOfFieldNode {
         _: &mut RenderGraphContext,
         render_context: &mut RenderContext<'w>,
         (
+            _depth_of_field,
```
Comment (rafalh) [inline crates/bevy_core_pipeline/src/dof/mod.rs:349]: not sure why do we need this and it seems like a hack to me

**Review by rafalh — COMMENTED** [review]
> @rafalh are you able to review this PR for me in turn? :) This is closely related to https://github.com/bevyengine/bevy/pull/15946.

@alice-i-cecile  I tried but I don't think I understand enough about the render graph to fully review this. I left some comments through. Most of changes looks okay to me.

**Review by tychedelia — APPROVED** [review]
Thanks for leaving the breadcrumbs to improve this later.

**Pull request merged**